src/unittests/f32_interval.spec.ts

```diff
@@ -2817,3 +2818,69 @@ g.test('crossInterval')
       `crossInterval([${x}], [${y}]) returned ${got}. Expected ${expected}`
     );
   });
+
+interface FaceForwardCase {
+  input: [number[], number[], number[]];
+  expected: IntervalBounds[][];
+}
+
+g.test('faceForwardIntervals')
+  .paramsSubcasesOnly<FaceForwardCase>(
+    // prettier-ignore
+    [
+      // vec2
+      { input: [[1.0, 0.0], [1.0, 0.0], [1.0, 0.0]], expected: [[[-1.0], [0.0]]] },
+      { input: [[-1.0, 0.0], [1.0, 0.0], [1.0, 0.0]], expected: [[[1.0], [0.0]]] },
+      { input: [[1.0, 0.0], [-1.0, 1.0], [1.0, -1.0]], expected: [[[1.0], [0.0]]] },
```

Comment:
Is this right? `[-1.0, 1.0]` and `[1.0, -1.0]` are at right angles to each other, so the dot is `0`.

The spec says:

> @const fn faceForward(e1: T, e2: T, e3: T) -> T
> Returns `e1` if `dot(e2, e3)` is negative, and `-e1` otherwise

As the dot is not negative, shouldn't this return `[-1.0, 0.0]`?
This is on the cusp of the fp-compare, so I wouldn't be surprised if you had to return either `-1.0` or `1.0` here.

Reply:
They are in a straight line aren't they? [-1, 1] is the top-left quad and [1, -1] is the bottom-right.
Also `dot([-1, 1], [1, -1]) = (-1 * 1) + (1 * -1) = -2`

Reply:
Sorry, my bad. Right you are.

Reply:
Do we have any cases for the dot(v1, v2) == 0?

Reply:
Having a 0 shouldn't be too much of an issue for normal inputs, because the calculation is using dot, which is mostly additions and multiplications, so composed of correctly rounded operations, so should be precise.

The last case, the one for subnormals, does have a spanning 0 situation, and demonstrates that both results are returned

Reply:
Just for completeness sake, please can we have a `[1.0, 0.0], [0.0, 1.0]` case?
I agree the subnormal case spans 0, but the fact they may be flushed means that these might not fully test the right-angle case.

Reply:
Done